hw3/03-react/game-of-thrones/src/views/Search.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { Component } from "react";
+import axios from "axios";
+import "../App.css";
+import TextField from "@mui/material/TextField";
+import Autocomplete from "@mui/material/Autocomplete";
+
+class Search extends Component {
+  state = {
+    characters: [],
+    filteredCharacter: {},
+  };
+  componentDidMount() {
+    axios.get(`https://thronesapi.com/api/v2/Characters`).then((res) => {
+      const characters = res.data;
+      let set = new Set();
+      let characterObjects = [];
+      characters.forEach((character) => {
+        if (!set.has(character.fullName)) {
+          let characterObject = {};
+          characterObject.label = character.fullName;
+          characterObject.imageURL = character.imageUrl;
+          characterObjects.push(characterObject);
+          set.add(character.fullName);
+        }
+      });
+      this.setState({ characters: characterObjects });
+    });
+  }
+
+  render() {
```

Comment:
This is a deprecated react function and is somewhat outdated, I would recommend using useEffect and useState in the future

Reply:
Thanks for the suggestion. I have replaced my entire code using useEffect and useState.